third_party/amd/include/Dialect/TritonAMDGPU/IR/TritonAMDGPUOps.td

```diff
@@ -21,15 +21,90 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-
 #ifndef TRITON_AMDGPU_OPS
 #define TRITON_AMDGPU_OPS
 
 include "mlir/IR/OpBase.td"
 include "mlir/IR/EnumAttr.td"
 include "mlir/Dialect/LLVMIR/LLVMOpBase.td"
 include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td" // Pure
+include "mlir/Interfaces/ViewLikeInterface.td" // OffsetSizeAndStrideOpInterface
 include "TritonAMDGPUDialect.td"
 include "TritonAMDGPUAttrDefs.td"
 
+class TritonAMDGPU_Op<string mnemonic, list<Trait> traits = []>
+    : Op<TritonAMDGPU_Dialect, mnemonic, traits>;
+
+//===----------------------------------------------------------------------===//
+// ViewSliceOp
+//===----------------------------------------------------------------------===//
+
+def TritonAMDGPU_ViewSliceOp
+    : TritonAMDGPU_Op<"view_slice", [AttrSizedOperandSegments,
+                                     OffsetSizeAndStrideOpInterface, Pure]> {
+  let summary = "view slice operation";
+  let description = [{
+    The "view_slice" operation enables "viewing" a slice of a tensor in
+    registers without data exchange.
+
+    The "view_slice" operation supports the following arguments:
+
+    * source: the base tensor on which to create a "view" tensor
+    * offsets: offsets into the base tensor at which to create the "view"
+    * size: size of the result "view" tensor
+    * strides: the number of strides for each dimension
+
+    Currently only 2D tensors are supported.
+
+    Example 1:
+
+    ```mlir
+    %1 = triton_gpu.convert_layout %0 : tensor<128x128x!tt.ptr<f16>, #blocked>
```

Comment:
Can you also be specific about the `#blocked` and `#blocked2` and mention the limitations on them?